std/experimental/color/conv.d

```diff
@@ -0,0 +1,523 @@
+// Written in the D programming language.
+
+/**
+    This module implements various _color type conversions.
+
+    Authors:    Manu Evans
+    Copyright:  Copyright (c) 2015, Manu Evans.
+    License:    $(WEB boost.org/LICENSE_1_0.txt, Boost License 1.0)
+    Source:     $(PHOBOSSRC std/experimental/color/conv.d)
+*/
+module std.experimental.color.conv;
+
+import std.experimental.color;
+import std.experimental.color.rgb;
+import std.experimental.color.xyz;
+
+import std.traits: isNumeric, isIntegral, isFloatingPoint, isSigned, TemplateOf;
+import std.typetuple: TypeTuple;
+
+
+/**
+Convert between color types.
+*/
+To convertColor(To, From)(From color) if(isColor!To && isColor!From)
```

Comment:
It's not practical for the else block to express the negation of all the other constraints. It's also not maintainable. It's definitely not readable. You can see how many there are already, there will be 3 more pairs of constraints added for every additional colour space. If there are 10 colour spaces, then there will be 30 pairs of constraints on the final function that simulates the else block.
The symbol names will be crazy long. I just don't see any compelling reason to break this. I did it this way for good reasons, you need to present better reasons than "i'd prefer it" to break it up. I can only see disadvantages to splitting it.
As it is, I concede it just looks a bit funny; instead of function headers, there is a static if at the head of each block, but I have tried to structure it as clearly as possible otherwise.